### DISCUSSION TITLE
How to get the type of a `tastyquery.ast.Trees.Tree`?

### DISCUSSION THREAD

**Discussion by civp** [scalacenter/tasty-query#32] [Q&A] [answered]
There's a member `tpe` in the class `Tree`, but the method `calculateType` doesn't seem to be implemented.

```scala
protected def calculateType(): Type = throw new TypeComputationError(this)
```

Is there currently a way to get the type of a `Tree`?

**Comment by bishabosha** [comment, marked as answer]
Hey, thank you for your question. Types still need to be implemented (as of February 2022).

One exception is the type of a `ValDef` tree, where that `ValDef` comes from the parameter of a class.

In [SymbolSuite.scala](https://github.com/scalacenter/tasty-query/blob/main/jvm/src/test/scala/SymbolSuite.scala), you could get the field of some class with this code:

```scala
test("get-type-of-parameter") {
  val ctx = getUnpicklingContext("simple_trees/ConstructorWithParameters")
  val theValPath = SimpleName("simple_trees") :: TypeName(SimpleName("ConstructorWithParameters")) :: SimpleName("theVal") :: Nil
  val theValSym = followPath(ctx.defn.RootPackage, theValPath)
  val theValType = theValSym.tree.tpe
}
```

**Reply by civp** [reply]
Thank you for your reply. I'll be looking forward to the implementation.
